test/CodeGen/AVR/sign-extension.ll

```diff
@@ -0,0 +1,11 @@
+; RUN: llc -march=avr < %s | FileCheck %s
+
+define i64 @sign_extended(i1) {
+; CHECK-LABEL: entry-block
+entry-block:
+  %1 = sext i1 %0 to i8
+  %2 = sext i8 %1 to i16
```

Comment:
This one's a humdinger:

```asm
sign_extended:                          ; @sign_extended
; BB#0:                                 ; %entry-block
	lsl	r24
	rol	r25
	lsl	r24
	rol	r25
	lsl	r24
	rol	r25
	lsl	r24
	rol	r25
	lsl	r24
	rol	r25
	lsl	r24
	rol	r25
	lsl	r24
	rol	r25
	lsl	r24
	rol	r25
	lsl	r24
	rol	r25
	lsl	r24
	rol	r25
	lsl	r24
	rol	r25
	lsl	r24
	rol	r25
	lsl	r24
	rol	r25
	lsl	r24
	rol	r25
	lsl	r24
	rol	r25
	asr	r25
	ror	r24
	asr	r25
	ror	r24
	asr	r25
	ror	r24
	asr	r25
	ror	r24
	asr	r25
	ror	r24
	asr	r25
	ror	r24
	asr	r25
	ror	r24
	asr	r25
	ror	r24
	asr	r25
	ror	r24
	asr	r25
	ror	r24
	asr	r25
	ror	r24
	asr	r25
	ror	r24
	asr	r25
	ror	r24
	asr	r25
	ror	r24
	asr	r25
	ror	r24
	movw	r20, r24
	asr	r21
	ror	r20
	asr	r21
	ror	r20
	asr	r21
	ror	r20
	asr	r21
	ror	r20
	asr	r21
	ror	r20
	asr	r21
	ror	r20
	asr	r21
	ror	r20
	asr	r21
	ror	r20
	asr	r21
	ror	r20
	asr	r21
	ror	r20
	asr	r21
	ror	r20
	asr	r21
	ror	r20
	asr	r21
	ror	r20
	asr	r21
	ror	r20
	asr	r21
	ror	r20
	movw	r18, r24
	movw	r22, r20
	movw	r24, r20
	ret
```

What general shape of tests would you like to see?